src/index/client/cache.ts

```diff
@@ -25,35 +25,89 @@
  */
 
 /**
- * @module openid/cache
+ * @module client/cache
  */
 
 import { createHash } from 'crypto';
 
-import { Environment } from '..';
+import { AuthClient, OAuth2Client } from './oauth2';
+import { OAuth2JWTClient } from './oauth2Jwt';
+import { OpenIdClient } from './openid';
+import { SalesforceClient } from './salesforce';
 
-import { OpenIdClient } from './client';
+const cache = new Map<string, AuthClient>();
+
+/**
+ * The Auth client environment parameters.
+ */
+export interface Environment {
+
+	/**
+	 * [Authorization Endpoint](https://tools.ietf.org/html/rfc6749#section-3.1).
+	 *
+	 * This must be defined for OAuth2 clients but should be omitted for OpenID clients.
+	 */
+	authorizationEndpoint?: string;
+
+	/**
+	 * The HTTP timeout used when creating an Auth Client.
+	 */
+	httpTimeout: number;
+
+	/**
+	 * The Issuer URI used when creating an Auth Client.
+	 */
+	issuerURI: string;
+
+	/**
+	 * [Revocation Endpoint](https://tools.ietf.org/html/rfc7009#section-2.1)
+	 *
+	 * This must be defined for OAuth2 clients but should be omitted for OpenID clients.
+	 */
+	revocationEndpoint?: string;
+
+	/**
+	 * [Token Endpoint](https://tools.ietf.org/html/rfc6749#section-3.2)
+	 *
+	 * This must be defined for OAuth2 clients but should be omitted for OpenID clients.
+	 */
+	tokenEndpoint?: string;
+
+	/**
+	 * The type of Auth Client.
+	 */
+	type: 'OAuth2' | 'OAuth2JWT' | 'OpenID' | 'Salesforce';
```

Comment:
Could be a string enum with a switch statement below when you decide what to create, potentially easier experience exporting the enum to use externally too. A mostly cosmetic change so don't mind if you want to leave as is.

Reply:
For now, I'll leave this as it is.